### DISCUSSION TITLE
Possible to bind an exchange to a queue during container boot/init

### DISCUSSION THREAD

**Discussion by John-Bowring** [docker-library/rabbitmq#609] [Q&A] [answered]
Basically I have a command to set up an bind between an exchange and a queue but it's erased every time I have to restart the container, is there any way to include this in the dockerfile or in the configuration. I tried adding the below to the init.sh but it did nothing.

`rabbitmqadmin declare binding source="amq.direct" destination_type="queue" destination="router-callback-queue" routing_key="router-callback-queue"`

**Comment by michaelklishin** [comment, marked as answer]
If applications cannot set up their own topology, use [boot time definition import](https://www.rabbitmq.com/docs/definitions#import-on-boot) instead of modifying image scripts.
